### DISCUSSION TITLE
Is it possible to add more shuffle options?

### DISCUSSION THREAD

**Discussion by stewday** [UnicornsOnLSD/finamp#597] [General]
Hi James & co, firstly, thanks for a great app!

I'd like to suggest two shuffle options:

- shuffle within a genre (this one is available in Jellyfin)
- shuffle all songs (don't think jellyfin has this)

Thanks again

**Comment by Chaphasilor** [comment]
Hi, there already is a shuffle all songs option, on the songs tab. It doesn't load all songs and then shuffle them (because that's a logistics issue for larger libraries), but fetches n songs (default is 250) in random order and plays those.

The implementation for shuffling genres is already partially there, so we could probably get that working fairly easily!

**Reply by stewday** [reply]
Thanks @Chaphasilor 

Couldn't see the songs shuffle for looking!
Of course it was there right in front of my nose - sorry about that.

Genre shuffle would be great - it has my vote.
